samples/vboxwrapper/vbox_common.h

```diff
@@ -97,161 +110,181 @@ struct VBOX_TIMESTAMP {
-    // amount of CPU time consumed by the VM (note: use get_vm_cpu_time())
-    double current_cpu_time;
-    // is the VM suspended?
-    bool suspended;
-    // is network access temporarily suspended?
-    bool network_suspended;
-    // is VM even online?
-    bool online;
-    // Is VM saving from checkpoint?
-    bool saving;
-    // Is VM restoring from checkpoint?
-    bool restoring;
-    // Has the VM crashed?
-    bool crashed;
-    // Has the VM been successfully started?
-    bool started_successfully;
-    // whether we were instructed to only register the VM.
-    // useful for debugging VMs.
-    bool register_only;
-    // for optional remote desktop; dynamically assigned
-    int rd_host_port;
-    bool headless;
-
-    int vm_pid;
-    int vboxsvc_pid;
+    public:
+        VBOX_BASE();
+        virtual ~VBOX_BASE();
+
+        std::string virtualbox_home_directory;
+        std::string virtualbox_scratch_directory;
+        std::string virtualbox_install_directory;
+        std::string virtualbox_guest_additions;
+        std::string virtualbox_version_raw;
+        std::string virtualbox_version_display;
+
+        FloppyIONS::FloppyIO* pFloppy;
+
+        // last polled copy of the log file
+        std::string vm_log;
+        // last VM guest log entry detected
+        VBOX_TIMESTAMP vm_log_timestamp;
+        // unique name for the VM
+        std::string vm_master_name;
+        // unique description for the VM
+        std::string vm_master_description;
+        // unique name for the VM or UUID of a stale VM if deregistering it
+        std::string vm_name;
+        // required CPU core count
+        std::string vm_cpu_count;
+        // name of the virtual machine disk image file
+        std::string image_filename;
+        // name of the virtual machine iso9660 disk image file
+        std::string iso_image_filename;
+        // name of the virtual machine cache disk image file
+        std::string cache_disk_filename;
+        // name of the virtual machine floppy disk image file
+        std::string floppy_image_filename;
+        // amount of CPU time consumed by the VM (note: use get_vm_cpu_time())
+        double current_cpu_time;
+        // is the VM suspended?
+        bool suspended;
+        // is network access temporarily suspended?
+        bool network_suspended;
+        // is VM even online?
+        bool online;
+        // Is VM saving from checkpoint?
+        bool saving;
+        // Is VM restoring from checkpoint?
+        bool restoring;
+        // Has the VM crashed?
+        bool crashed;
+        // Has the VM been successfully started?
+        bool started_successfully;
+        // whether we were instructed to only register the VM.
+        // useful for debugging VMs.
+        bool register_only;
+        // for optional remote desktop; dynamically assigned
+        int rd_host_port;
+        bool headless;
+
+
+        std::streamoff log_pointer;
+        std::string state;
```

Comment:
The state is changed only if new lines are added to the log. State holds the last read state even if no new lines are added to the log.

Reply:
Apart from initialisation, `state` is accessed only in `VBOX_BASE::read_vm_log`. Since it's used as a temporary variable I don't see the need to have it declared a class member variable.

State string is already held in `static string vmstate_old` in `VBOX_VM::poll`. If you want to have the last read state string available outside of `VBOX_VM::poll` you could a bit of refactoring. I don't think it's useful to have two variables for the same thing.